Sources/StreamChatUI/ChatMessageList/ChatChannelHeaderView.swift

```diff
@@ -0,0 +1,144 @@
+//
+// Copyright © 2021 Stream.io Inc. All rights reserved.
+//
+
+import StreamChat
+import UIKit
+
+/// The view that displays channel information on the message list header
+public typealias ChatChannelHeaderView = _ChatChannelHeaderView<NoExtraData>
+
+/// The view that displays channel information on the message list header
+open class _ChatChannelHeaderView<ExtraData: ExtraDataTypes>:
+    _View,
+    ThemeProvider,
+    _ChatChannelControllerDelegate {
+    /// Controller for observing data changes within the channel.
+    open var channelController: _ChatChannelController<ExtraData>?
+
+    /// The user id of the current logged in user.
+    open var currentUserId: UserId? {
+        channelController?.client.currentUserId
+    }
+
+    /// Timer used to update the online status of member in the channel.
+    open var timer: Timer?
+
+    /// The amount of time it updates the online status of the members.
+    /// By default it is 60 seconds.
+    open var statusUpdateInterval: TimeInterval { 60 }
+
+    /// A view that displays a title label and subtitle in a container stack view.
+    open private(set) lazy var titleContainerView: TitleContainerView = components
+        .titleContainerView.init()
+        .withoutAutoresizingMaskConstraints
+
+    override open func setUp() {
+        super.setUp()
+
+        channelController?.setDelegate(self)
+    }
+
+    override open func setUpLayout() {
+        super.setUpLayout()
+
+        embed(titleContainerView)
+    }
+
+    override open func updateContent() {
+        super.updateContent()
+
+        titleContainerView.content = (titleText, subtitleText)
+
+        makeTimer()
```

Comment:
hm, why we create a timer in `updateContent` and not in `setUp`?

Reply:
Yes, you are right, nice catch! This was because before the timer was dependent on `is.directMessageChannel` and that could change after the `updateContent`, right now it doesn't make sense to be in `updateContent` and indeed it can go to setUp, will change 👍

Reply:
Done ✅